src/main/java/seedu/address/logic/parser/AddWeddingCommandParser.java

```diff
@@ -0,0 +1,46 @@
+package seedu.address.logic.parser;
+
+import static seedu.address.logic.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_DATE;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_NAME;
+
+import java.time.LocalDate;
+import java.util.stream.Stream;
+
+import seedu.address.logic.commands.AddWeddingCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.wedding.Wedding;
+
+/**
+ * Parses input arguments and creates a new AddWeddingCommand object
+ */
+public class AddWeddingCommandParser implements Parser<AddWeddingCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the AddWeddingCommand
+     * and returns an AddWeddingCommand object for execution.
+     * @throws ParseException if the user input does not conform to the expected format
+     */
+    public AddWeddingCommand parse(String args) throws ParseException {
+        ArgumentMultimap argMultimap =
+                ArgumentTokenizer.tokenize(args, CliSyntax.PREFIX_NAME, CliSyntax.PREFIX_DATE);
+        if (!arePrefixesPresent(argMultimap, CliSyntax.PREFIX_NAME, CliSyntax.PREFIX_DATE)
+                || !argMultimap.getPreamble().isEmpty()) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, AddWeddingCommand.MESSAGE_USAGE));
+        }
+        argMultimap.verifyNoDuplicatePrefixesFor(PREFIX_NAME, PREFIX_DATE);
+        String name = argMultimap.getValue(CliSyntax.PREFIX_NAME).get();
+        LocalDate date = ParserUtil.parseDate(argMultimap.getValue(CliSyntax.PREFIX_DATE).get());
+
+        Wedding wedding = new Wedding(name, date.toString());
```

Comment:
Also I think cos we are storing wedding name as a string in the wedding object and not the name object it does not check if the name field is left blank so I am still able to add a wedding with an empty name which should be a bug [(addw n/ d/12/12/2012) is executed without any issues though it should be returning an error message]